vcd/resource_vcd_snat.go

```diff
@@ -119,23 +124,44 @@ func resourceVcdSNATCreate(d *schema.ResourceData, meta interface{}) error {
 func resourceVcdSNATRead(d *schema.ResourceData, meta interface{}) error {
 	vcdClient := meta.(*VCDClient)
 
-	e, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
 	if err != nil {
 		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
 	}
 
+	// Terraform refresh won't work if Rule was edit in UI. vCD API uses tag elements to map edge gtw Id's
+	// and UI will reset the tag element on update.
```

Comment:
Same thing on note about removed object as in DNAT.

Reply:
updated